libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/fakeResolver.kt

```diff
@@ -222,31 +237,38 @@ private fun buildFieldOfNonNullType(
   }
 }
 
-class DefaultFakeResolver(val types: List<CompiledNamedType>) : FakeResolver {
-  private var currentInt = 0
-  private var currentFloat = 0.0
-  private var currentBoolean = false
-
-  private var impl = mutableMapOf<String, Int>()
+/**
+ * A [FakeResolver] that generates:
+ * - values based on the object id hashcode for Int/Float/Boolean/ID
+ * - values based on the field name for strings
+ *
+ * For object id, [DefaultFakeResolver] uses `@typePolicy` if present. Or you can also override it by setting the
+ * "__stableId" property:
+ *
+ * ```kotlin
+ * val cat = buildCat {
+ *   this["stableId"] = "foo"
+ * }
+ * ```
+ */
+open class DefaultFakeResolver(types: List<CompiledNamedType>) : FakeResolver {
+  private val enumTypes = types.filterIsInstance<EnumType>()
+  private val allTypes = types
 
   override fun resolveLeaf(context: FakeResolverContext): Any {
     return when (val name = context.mergedField.type.rawType().name) {
-      "Int" -> currentInt++
-      "Float" -> currentFloat++
-      "Boolean" -> (!currentBoolean).also { currentBoolean = it }
+      "Int" -> context.id.hashCode() % 100
+      "Float" -> (context.id.hashCode() % 100000).toFloat() / 100.0
+      "Boolean" -> context.id.hashCode() % 2 == 0
       "String" -> {
```

Comment:
Did you consider adding `context.id.hashCode()` to the default strings?

That might be nice because it means that the generated strings are always unique to the object, so it might help to avoid some false negatives? e.g. I think [this assertion](https://github.com/apollographql/apollo-kotlin/pull/4468/files#diff-686a647ac7e0eb49657d51dd021f92adfae0b63e9a3540b668fb0e59ffdc507aR267) will always pass as it is now, even if there were two distinct objects there.

But, maybe it has other downsides that I'm not thinking of?

Reply:
Oh I meant more like `user.name = "name_8756512089"`, but yeah I'm also not sure if it's worth it anyway.

Reply:
Ah, I see, makes a lot of sense 👍. I'm a bit reluctant to introduce this now cause it might break existing tests but will keep it in mind for future evolutions.